### PR TITLE
Add service for newsletter signup

### PR DESCRIPTION
**Pull request by ludeeus**

### CODE REVIEW

File: src/services/newsletter.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  NOT_VALID = "not_valid",
+  SUBSCRIPTION = "subscription",
+}
+
+export async function newsletterHandler(
+  requestUrl: URL,
+  request: Request,
+  sentry: Toucan
+): Promise<Response> {
+  if (
```
Comment (balloob) [inline src/services/newsletter.ts:18]: Is there not a router helper for workers that can handle with this?

unknown path -> 404 Not Found
unsupported method -> 405 Method Not Supported

File: src/services/newsletter.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  });
+
+  if (!response.ok) {
+    throw new ServiceError(
+      "Could not subscribe",
+      NewsletterErrorType.SUBSCRIPTION
+    );
+  }
+
+  return new Response(null, {
```
Comment (balloob) [inline src/services/newsletter.ts:59]: If we're not going to include a response, why even bother with formdata and not just accept a JSON request?

File: src/services/newsletter.ts (hunk 1 above)
Comment (ludeeus) [reply src/services/newsletter.ts:18]: What each service support is up to each service
But I can adjust this to throw 405 if not POST

File: src/services/newsletter.ts (hunk 2 above)
Comment (ludeeus) [reply src/services/newsletter.ts:59]: Your idea was to call this from an embedded element?
Which then would use form data?
We can return whatever we want here

File: src/services/newsletter.ts (hunk 1 above)
Comment (balloob) [reply src/services/newsletter.ts:18]: oh right, it's just 1 endpoint.

In the frontend we have a router at every level and each level forwards to a different file. Just keeps it simple.

Comment (ludeeus) [reply src/services/newsletter.ts:18]: We could do that here as well, but so far each service does 1 thing

File: src/services/newsletter.ts (hunk 2 above)
Comment (balloob) [reply src/services/newsletter.ts:59]: Well if we have a form on a page, we can either add some JavaScript to handle the signup or submit it as an HTML form. With JSON we can have the JavaScript render a success message. But if we use an HTML form submit, we should show _something_ and not a blank page.

File: src/services/newsletter.ts (hunk 1 above)
Comment (balloob) [reply src/services/newsletter.ts:18]: Yeah but I'm just thinking about the future. This repo will contain more things

File: src/services/newsletter.ts (hunk 2 above)
Comment (ludeeus) [reply src/services/newsletter.ts:59]: Whatever you want/need we can do, JSON body is definitely the simplest to implement here.

Comment (ludeeus) [reply src/services/newsletter.ts:59]: Ah, ok, I get it.

Comment (ludeeus) [inline src/services/newsletter.ts:59]: How about:
```
return new Response("You are now subscribed to the Home Assistant Newsletter 🎉", {
```

Comment (balloob) [reply src/services/newsletter.ts:59]: If it's consumed from an HTML form, we should make it look a little better. Not much, but at least a "Go back" button that triggers `history.back()`?

Comment (balloob) [reply src/services/newsletter.ts:59]: or maybe it's meant to open in a new tab and we trigger `window.close()`

**Review by balloob — APPROVED** [review]

**Pull request merged**